docs/source/app-dev/bindings-java/codegen.rst

```diff
@@ -380,3 +390,99 @@ The Java code generated for this variant is:
     public Published(Long year, String publisher) { /* ... */ }
     public Record toValue() { /* ... */ }
   }
+
+Parameterized types
+~~~~~~~~~~~~~~~~~~~
+
+A :ref:`parameterized type <daml-ref-parameterized-types>` is where the actual type is specified at instantiation as *type parameters* and not as part of type definition.
+The Java Code Generator uses Java Generic types to represent the DAML parameterized type.
+
+Below is a DAML fragment defining the parameterized type `Attribute` for use by the `
+
+.. code-block:: daml
+  :caption: Com/Acme.daml
+
+  daml 1.2
+  module Com.Acme where
+
+  data Attribute a = Attribute
+      with v : a
+
+  data BookAttributes = BookAttributes with
+     pages : (Attribute Int)
+     authors : (Attribute [Text])
+     title : (Attribute Text)
+
+A file Java file is generated for the `Attribute` data type that defines the Java Generic class:
+
+Note: If the parameterized type is contained in a type where the *actual* type is specified (as in the `BookAttributes` type above), then the serialization
+and deserialization of the enclosing type provides the necessary methods for serialization and deserialization.
+
+.. code-block:: java
+  :caption: com/acme/Attribute.java
+  :emphasize-lines: 3,8,10
+
+  package com.acme;
+
+  public class Attribute<a> {
+    public final a value;
+
+    public Attribute(a value) { /* ... */  }
+
+    public Record toValue(Function<a, Value> toValuea) { /* ... */ }
+
+    public static <a> Attribute<a> fromValue(Value value$, Function<Value, a> fromValuea) { /* ... */ }
+  }
+
+**Serializing**
+
+To serialize an instance of the `Attribute<a>` data type function for creating the Ledger API equivalent of the attribute value
+is passed to the `toValue` method as the `fromValuea` argument (see the above `com/acme/Attribute.java` source extract).
+
+Below is a Java fragment to serialize an attribute with a `java.lang.Long` value to Ledger API representation using the *method reference*
+`Int64::new` to create a new instance of the Java Bindings value type.
+
+.. code-block:: java
+
+  Attribute<Long> pagesAttribute = new Attributes<>(42L);
+
+  Value serializedPages = pagesAttribute.toValue(Int64::new);
+
+See :ref:`DAML To Java Type Mapping <daml-codegen-java-primitive-types>` for the Java Bindings value types need to be created in the `fromValue` method.
+
+Note: That if the DAML type is a record that has more that one parameterized field, then a function for creating the
```

Comment:
I think we should replace `field` with type here (and the line below). Two fields of type `a` still only need one function `a => Value`.

Reply:
Done